cranelift/codegen/src/isa/pulley_shared/inst.isle

```diff
@@ -168,6 +168,14 @@
   )
 )
 
+(decl pure partial amode_add (Amode i32) Amode)
+(rule (amode_add (Amode.RegOffset base offset) amt)
```

Comment:
Perhaps a comment here noting that we control `offset` (constants in our lowering rules) so we don't have to worry about the case where the `if-let` fails?

Reply:
Actually -- we're adding to a user-controlled initial offset -- should we have a generic fallback here?

Reply:
This is only used for i128 stores which is only used by Cranelift trampolines in tests, and technically it's universally correct just perhaps not as useful as it could be in the sense that it'll fail if the offset addition overflows in the future. Basically I wasn't sure if it was worth trying to be 100% correct here when nothing known will hit this, so I opted for a "will loudly fail" in the future rather than filling out all the particulars to get this working optimally. I'll look at it again though and see if I can't think of an always-works solution.